src/ooo_cpu.cc

```diff
@@ -1786,19 +1767,19 @@ void O3_CPU::reg_RAW_release(uint32_t rob_index)
                         ROB.entry[i].scheduled = COMPLETED;
 
 #ifdef SANITY_CHECK
-                        if (RTE0[RTE0_tail] < ROB_SIZE)
+                        if (ready_to_execute[ready_to_execute_tail] < ROB_SIZE)
                             assert(0);
 #endif
                         // remember this rob_index in the Ready-To-Execute array 0
-                        RTE0[RTE0_tail] = i;
+                        ready_to_execute[ready_to_execute_tail] = i;
 
                         DP (if (warmup_complete[cpu]) {
-                        cout << "[RTE0] " << __func__ << " instr_id: " << ROB.entry[i].instr_id << " rob_index: " << i << " is added to RTE0";
-                        cout << " head: " << RTE0_head << " tail: " << RTE0_tail << endl; }); 
+                        cout << "[ready_to_execute] " << __func__ << " instr_id: " << ROB.entry[i].instr_id << " rob_index: " << i << " is added to ready_to_execute";
+                        cout << " head: " << ready_to_execute_head << " tail: " << ready_to_execute_tail << endl; }); 
 
-                        RTE0_tail++;
-                        if (RTE0_tail == ROB_SIZE)
-                            RTE0_tail = 0;
+                        ready_to_execute_tail++;
+                        if (ready_to_execute_tail == ROB_SIZE)
+                            ready_to_execute_tail = 0;
```

Comment:
Could you check that this does not bypass the scheduler width? This section is very similar to L916-924.

Reply:
I agree.  I'll see if I can make it so that we only add to ready_to_execute[] in do_scheduling().  reg_RAW_release() should only mark the dependencies as resolved, not also do the scheduling.

Reply:
If you think those changes should be in a separate PR, then go ahead and merge this. I like it overall.

Reply:
I was able to make it so that instructions that have had their dependencies resolved need to go through scheduling again, but now that I've thought about this more, I think I like this original behavior better.  Allowing instructions to execute as soon as possible after their dependencies are resolved is kind of like a bypass network.  And since they don't have have their executed field set to anything, they still count as being in the scheduler, so they are still taking up scheduler width anyway.

I like it better this way, so I think this should be merged as-is, but I think we should eventually replace the circular buffer with something like a priority queue to make the oldest instructions execute first.